### ISSUE TITLE
GroupBox doesn't apply ForeColor in ambient condition

### ISSUE THREAD

**Issue by KlausLoeffelmann** [dotnet/winforms#9958]
### .NET version

.NET 6.

### Did it work in .NET Framework?

Not tested/verified

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No.

### Issue description

When the GroupBox is inheriting its ForeColor by its Parent control, the color is not correctly applied.

### Steps to reproduce

* Create a new WinForms Form.
* Insert a TableLayoutPanel
* Insert a GroupBox
* Change the Backcolor of the Form to 32,32,32
* Change the ForeColor of the Form to 240,240,240

The GroupBox should render the Text in light on dark background, but only the Frame is light. The text gets rendered in the original color:

![image](https://github.com/dotnet/winforms/assets/9663150/d3aafcca-082d-47b0-a245-3ca9ce01d46e)

@Olina, can you test, if this is a Framework regression? Thanks!

**Comment by elachlan** [comment]
> can you test, if this is a Framework regression? Thanks!

@Olina-Zhang

**Comment by Olina-Zhang** [comment]
It isn't a regression issue, .Net framwork and .Net 8.0 have the same behavior. When setting Form's BackColor with dark value and ForeColor with light value, the added GroupBox has same value of BackColor and ForeColor, but the text is dark. We need to set ForeColor with same value again, it works well. 
![BackColor_ForeColor](https://github.com/dotnet/winforms/assets/26474449/c7dc2057-5908-462d-8557-1dab3d03b1e3)

**Comment by KlausLoeffelmann** [comment]
It doesn't work in certain situations, though, where the Designer notices an ambient condition already, and the Serializer doesn't spill out code for that, because at that point it "knows" that the Property is the same as the assigned parent, and so doesn't see the need to serialize it because the `ShouldSerializeForeColor` in that setup returns `false`. And then it's impossible to set.

This is really bad for Dark Mode scenarios, and we would need to fix this rather sooner than later!

**Comment by KlausLoeffelmann** [comment]
@elachlan, OT: Could you send me an email at klaus ';dot' loeffelmann 'at;' microsoft '.' com so I could follow-up with you on an issue? Thanks! (And thanks for correcting the @ to Olina here!!)

**Comment by elachlan** [comment]
@KlausLoeffelmann email is sent.

**Comment by memoarfaa** [comment]
@KlausLoeffelmann 

Why not set it up manually?

```
namespace TestGroupBox
{
    public class MyGroupBox : GroupBox
    {

        public MyGroupBox() 
        {
            //We need to set it here
            ForeColor =  base.ForeColor;  // SystemColors.ControlText;    
        } 
      

        protected override void OnParentChanged(EventArgs e)
        {
            base.OnParentChanged(e);
           // if ForeColor need to be changed if the Parent ForeColor changed
            Form parentfrm = this.FindForm();
            if (parentfrm != null && this.ForeColor == SystemColors.ControlText)
                parentfrm.ForeColorChanged += delegate { ForeColor = parentfrm.ForeColor;};
        }

     
        protected override void OnPaint(PaintEventArgs e)
        {

            if (this.FindForm() != null && this.ForeColor == SystemColors.ControlText)
                ForeColor = this.FindForm().ForeColor;
            base.OnPaint(e);
        }
       


    }
}
```

I made quick design  time debug session and it work fine 


https://github.com/dotnet/winforms/assets/12494184/1da159d9-f5cb-4157-8d72-899651b7503c

**Comment by KlausLoeffelmann** [comment]
I am not sure what you mean. This property/properties being ambient, should not make any additional effort to pick colors up necessary. That's why I consider this to be a bug, and we should fix it.

**Comment by Takzzg** [comment]
Sorry for necroposting, but I just stumbled upon this very issue. First time using winforms too, so please excuse my lack of terminology.
I'm mostly sharing this in case it helps someone

TLDR; set the GroupBox's ForeColor to its parent's ForeColor inside the OnLoad() function, and again whenever ForeColor should change.

I'm working on a C# .Net Framework 4.7.2 app. 
My form consists of a split container with buttons on Panel1, and an empty Panel2. Whenever a button in Panel1 is clicked, Panel2 will show the respective UserControl for that button (basically changing tabs).

One of this UserControls (the "Options" tab) has a GroupBox with two buttons in it, for changing the Form's ForeColor and BackColor respectively. Whenever the user changes one of this options, an event is raised from the Options UserControl and handled in the Form, which takes the new color and applies it to the Form, with all other Controls inheriting this value.

This does not apply to the GroupBox's title ForeColor, which is always black, even when nested controls are using the inherited values. 
I figured I can loop over all GroupBoxes and set their ForeColor whenever the ForeColor option changes. This worked until I tried going to another tab and back to the Options tab, which resets the title color back to black,.

Thought I could fix this by setting all GroupBoxes ForeColor inside the UserControl's OnLoad() function, instead of when the ForeColor option changes, but clearly this means I have to change tabs in order to see the results.

So my Band-Aid solution for the moment is doing both. Read parent's color on load, loop over all GroupBoxes, and set their ForeColor, and whenever ForeColor is changed from Options, loop again and reapply. Feels weird having to do it this way, but oh well...

**Comment by Takzzg** [comment]
Here's what i ended up with, based on @memoarfaa's answer.
C# .Net Framework v4.7.2 desktop app (WinForms)

```cs 
using System;
using System.Windows.Forms;

namespace <Project_Name>.controls {
    public partial class Styled_GroupBox : GroupBox {
        public Styled_GroupBox() {
            InitializeComponent();
            ForeColor = default;
        }

        protected override void OnParentForeColorChanged(EventArgs e) {
            base.OnParentForeColorChanged(e);
            ForeColor = Parent.ForeColor;
        }
    }
}

```
